sources/packages/backend/libs/integrations/src/esdc-integration/msfaa-integration/msfaa-integration.service.ts

```diff
@@ -143,19 +142,33 @@ export class MSFAAIntegrationService {
   /**
    * Get the list of all response files waiting to be downloaded from the
    * SFTP filtering by the the regex pattern '/PE*\.txt/i'.
+   * @param offeringIntensity offering intensity.
    * @returns full file paths for all response files present on SFTP.
    */
-  async getResponseFilesFullPath(): Promise<string[]> {
+  async getResponseFilesFullPath(
+    offeringIntensity: OfferingIntensity,
+  ): Promise<string[]> {
     let filesToProcess: Client.FileInfo[];
     const client = await this.getClient();
     try {
-      filesToProcess = await client.list(
-        `${this.esdcConfig.ftpResponseFolder}`,
-        new RegExp(
-          `^${this.esdcConfig.environmentCode}EDU.PBC.MSFA.REC.*`,
-          "i",
-        ),
-      );
+      if (offeringIntensity === OfferingIntensity.fullTime) {
```

Comment:
Please use the `if` to define one or another regex only, it will simplify the code.